puppet/lib/puppet/indirector/catalog/puppetdb.rb

```diff
@@ -214,7 +214,7 @@ def sort_unordered_metaparams(hash)
         params = resource['parameters']
         UnorderedMetaparams.each do |metaparam|
           if params[metaparam].kind_of? Array then
-            values = params[metaparam].sort
+            values = params[metaparam].sort_by {|x| x.to_s}
```

Comment:
discussed in channel, but it's needed for the checksum computation we do before command submission